tests/cbmc/proofs/s2n_hash_free/s2n_hash_free_harness.c

```diff
@@ -23,15 +23,42 @@ void s2n_hash_free_harness()
     /* Non-deterministic inputs. */
     struct s2n_hash_state *state = cbmc_allocate_s2n_hash_state();
 
+    /* Assumptions. */
+    __CPROVER_assume(s2n_result_is_ok(s2n_hash_state_validate(state)));
+
+    struct rc_keys_from_hash_state saved_hash_state;
+    save_rc_keys_from_hash_state(state, &saved_hash_state);
+
     /* Operation under verification. */
-    if (s2n_hash_free(state) == S2N_SUCCESS)
-    {
-        /* Post-conditions. */
-        if (state != NULL) {
-            assert(state->hash_impl->free != NULL);
-            assert(IMPLIES(s2n_is_in_fips_mode(), state->digest.high_level.evp.ctx == NULL));
-            assert(IMPLIES(s2n_is_in_fips_mode(), state->digest.high_level.evp_md5_secondary.ctx == NULL));
-            assert(state->is_ready_for_input == 0);
+    int result = s2n_hash_free(state);
+
+    /* Post-conditions. */
+    assert(result == S2N_SUCCESS);
```

Comment:
Nit: We don't need to store result.
```c
assert(s2n_hash_free(state) == S2N_SUCCESS);
```
If you think it's cleaner to gather all the post-conditions in their own block, you can keep the current code.

Reply:
Thanks, changed as suggested.